cmd/bpfman-operator/main.go

```diff
@@ -129,10 +156,26 @@ func main() {
 		Scheme: mgr.GetScheme(),
 	}
 
+	setupLog.Info("Discovering APIs")
+	dc, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(err, "can't instantiate discovery client")
+		os.Exit(1)
+	}
+
+	isOpenshift, err := getPlatformInfo(dc, mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(err, "unable to determine platform")
+		os.Exit(1)
```

Comment:
Is it likely this could fail? I guess call to KubeAPI server fails. Wondering of we should bail on a failure or not?

Reply:
Usually if ANY setup fails in we bail here, I think the step before `mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{` would fail if for some reason the process couldn't access the API server